src/components/TimerMode.jsx

```diff
@@ -21,14 +19,16 @@ const Timer = ({ duration, status, uid, docid }) => {
     }
   };
 
-  const updateDuration = async (seconds) => {
-    await updateDoc(taskDocRef, {
-      duration: seconds,
-      isActive,
-    });
-  };
-
-  updateDuration(seconds);
+  useEffect(() => {
+    const taskDocRef = doc(db, `users/${uid}/tasks`, `${docid}`);
+    const updateDuration = async (seconds) => {
+      await updateDoc(taskDocRef, {
```

Comment:
Don't set the duration on every tick of the timer. That's expensive. Better to move the saving of timer duration on a click action.